infra/main.tf

```diff
@@ -15,6 +15,34 @@ resource "aws_elasticache_subnet_group" "shared-redis-cache-subnet-group" {
   ]
 }
 
+resource "aws_elasticache_user" "web-api-lambda-user" {
+  user_id       = "web-api-lambda-user"
+  user_name     = "web-api-lambda-user"
+  access_string = "on ~* -@all +@connection +@fast +@slow +@scripting +@read +@write"
+  engine        = "REDIS"
+  passwords     = [var.lambda_dotnet_redis_user_password, var.lambda_dotnet_redis_user_password]

```

Comment:
Possibly a typo? You have the same password being passed twice

Reply:
[based of this](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/elasticache_user#authentication_mode-2), i think you can place this in the `authentication_mode` block to keep things close to the auth type definition

Reply:
oh interesting...

Reply:
let me try it out, the reason with two passwords of the same is that there are two passwords you can set for a user, just being explicit about it!